merkle/map_verifier.go

```diff
@@ -42,23 +42,44 @@ func VerifyMapInclusionProof(treeID int64, index, leafHash, expectedRoot []byte,
 		}
 	}
 
+	var runningHash []byte
+	if len(leaf) != 0 {
+		runningHash = h.HashLeaf(treeID, index, leaf)
+	}
+
 	nID := storage.NewNodeIDFromHash(index)
+	for height, sib := range nID.Siblings() {
+		pElement := proof[height]
 
-	runningHash := make([]byte, len(leafHash))
-	copy(runningHash, leafHash)
+		// Since HashChildren(e0, e0) is not always equal to the empty
```

Comment:
How's this: 
// Since empty values are tied to a location and a 
// HashEmpty(leve1) != HashChildren(E0, 
// Therefore we need to maintain an empty marker along the
// proof path until the first non-empty element so we can call
// HashEmpty once at the top of the empty branch.